h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstGroup.java

```diff
@@ -797,14 +917,19 @@ public String toString() {
   // here.
   private static class BuildGroup extends MRTask<BuildGroup> {
     final int[] _gbCols;
+    final GroupByColType[] _gbColsTypes;
+    final IcedHashMap<GroupByColType, Long> _gbColTypesCounts;
```

Comment:
this would imply there is some flexibility to the contents of this hashmap when in fact its only used to store two keys and never iterated over, I think having two fields would end up in less code

Reply:
True. I guess I was overthinking this one. It seems that there won't be any other types necessary. I changed it.